salt/metalk8s/kubeadm/init/kubelet-start/configured.sls

```diff
@@ -0,0 +1,62 @@
+{%- from "metalk8s/map.jinja" import kubelet with context %}
+
+Create kubelet service environment file:
+  file.managed:
+    - name: {{ kubelet.service.environment_file }}
+    - source: salt://metalk8s/kubeadm/init/kubelet-start/files/kubeadm.env
+    - template: jinja
+    - user: root
+    - group: root
+    - mode: 644
+    - context:
+        options: {{ kubelet.service.options }}
+
+{#- Default dataset for kubelet config #}
+{%- set dataset = salt['grains.filter_by']({
+  'default': {
+    'kind': 'KubeletConfiguration',
+    'authentication': {
```

Comment:
I'm a bit surprised these (and `authorization` below) are required to be specified. I seem to remember they were not before.

Reply:
With my test with kubelet1.13 it's required and also if I only enable anonymous but not disable webhook  authentication that doesn't work.
And for authorization it's seems that it's a known bug https://github.com/kubernetes/kubernetes/issues/68939 should be AlwaysAllow by default but it's not

Reply:
Ah. So, the reason things were 'different' in my testing: I wasn't using a config file at all, but passing required options (actually, only the manifest path) on the command-line.

Given this, the state is now not 'idempotent' with what's going to happen later on, when we create the 'real' `kubelet` configuration file. If we'd re-run the state, we'll end up replacing said 'real' config with this minimal one again, and replace with the 'real' later, etc.

Maybe this file should only be generated with this config if it does not already exist, or rather, it doesn't specify the manifest path properly, and we don't really care about other settings?

Reply:
Yes I agree with file.serialize we can merge also and it's maybe make sense for `staticPodPath` but the problem is the authentication and authorization stuff because for kind apiversion and staticPodPath we could always merge I think.

But basically if we want a running kubelet at this step we will have the "not idempotent" somewhere.

Reply:
We can maybe have 2 service files for kubelet
one who run kubelet wit no webhook
and then we create another real service config file that reset the args of kubelet

Reply:
if we want to achieve idempotency, the only solution I can think of right now is to have a "temporary" kubelet service file generated only if the "final" kubelet service is not here.
This way, the first run is indeed taking 2 steps to reach the final kubelet state, but in case of upgrade or rerun, we have only one, idempotent state

Reply:
Yes was my previous comment just with an only if.
Ok so I think we will do it in a latter PR at the `kubeconfig kubelet` kubeadm phase I think, so we know what exactly needed